### PR TITLE
feat: log detection of config file

### LINKED ISSUE #14578: Notify when configuration file is found
Sometimes I'm not sure my `deno.json` or `deno.jsonc` file was auto detected (sometimes a typo, others times I haven't created it yet). So I'm proposing for the `Deno run` and other [tools](https://deno.land/manual/tools) to notify if it was found.

```
Watcher File change detected! Restarting!
Config File found at ./deno.jsonc
Check file:///C:/path/to/file.ts
```

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #14578

### CODE REVIEW

File: cli/args/config_file.rs [hunk 1]
```diff
@@ -524,8 +526,9 @@ impl ConfigFile {
       if checked.insert(ancestor.to_path_buf()) {
         for config_filename in CONFIG_FILE_NAMES {
           let f = ancestor.join(config_filename);
-          match ConfigFile::read(f) {
+          match ConfigFile::read(&f) {
             Ok(cf) => {
+              info!("{} found at '{}'", colors::intense_blue("Config File"), f.display());
```
Comment (bartlomieju) [inline cli/args/config_file.rs:531]:
```suggestion
              info!("{} found at '{}'", colors::intense_blue("Config file"), f.display());
```

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/args/config_file.rs [hunk 2]
```diff
@@ -524,8 +526,13 @@ impl ConfigFile {
...
         for config_filename in CONFIG_FILE_NAMES {
           let f = ancestor.join(config_filename);
-          match ConfigFile::read(f) {
+          match ConfigFile::read(&f) {
             Ok(cf) => {
+              info!(
+                "{} found at '{}'",
+                colors::intense_blue("Config file"),
+                f.display()
+              );
```
Comment (dsherret) [inline cli/args/config_file.rs:535]: Please let's not do this. Can we instead make this a `--log-level=debug`?

Comment (bartlomieju) [reply cli/args/config_file.rs]: Oh shoot 🤦 sorry, I missed that it's not using `log::info`

Comment (crowlKats) [reply cli/args/config_file.rs]: well, its not now, before it was. should we make it print to log::info for watch flag, or just have it everywhere behind debug?

**Review by dsherret — APPROVED** [review]
LGTM too

**Pull request merged**